public/app/features/alerting/unified/components/rule-editor/QueryRows.tsx

```diff
@@ -218,13 +227,18 @@ function copyModel(item: AlertQuery, settings: DataSourceInstanceSettings): Omit
   };
 }
 
-function newModel(item: AlertQuery, settings: DataSourceInstanceSettings): Omit<AlertQuery, 'datasource'> {
+async function newModel(
+  item: AlertQuery,
+  settings: DataSourceInstanceSettings
+): Promise<Omit<AlertQuery, 'datasource'>> {
+  const ds = await getDataSourceSrv().get(item.datasourceUid);
```

Comment:
Maybe we could move this line to `onDataSourceChange` function, and instead of passing `settings` we could pass the data source object. Thanks to that this function would be still simple and wouldn't require async calls

Reply:
Yes we can do it 👌
We still need to pass settings, though.